mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_token_test.go

```diff
@@ -0,0 +1,100 @@
+package resourcemanager_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestEphemeralServiceAccountToken_basic(t *testing.T) {
+	t.Parallel()
+
+	serviceAccount := envvar.GetTestServiceAccountFromEnv(t)
+	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "basic", serviceAccount)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acctest.AccTestPreCheck(t) },
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEphemeralServiceAccountToken_basic(targetServiceAccountEmail),
+			},
+		},
+	})
+}
+
+func TestEphemeralServiceAccountToken_withDelegates(t *testing.T) {
+	t.Parallel()
+
+	serviceAccount := envvar.GetTestServiceAccountFromEnv(t)
+	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "delegates", serviceAccount)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acctest.AccTestPreCheck(t) },
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEphemeralServiceAccountToken_withDelegates(targetServiceAccountEmail),
+			},
+		},
+	})
+}
+
+func TestEphemeralServiceAccountToken_withCustomLifetime(t *testing.T) {
+	t.Parallel()
+
+	serviceAccount := envvar.GetTestServiceAccountFromEnv(t)
+	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "lifetime", serviceAccount)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acctest.AccTestPreCheck(t) },
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEphemeralServiceAccountToken_withCustomLifetime(targetServiceAccountEmail),
+			},
+		},
+	})
+}
+
+func testAccEphemeralServiceAccountToken_basic(serviceAccountEmail string) string {
+	return fmt.Sprintf(`
+ephemeral "google_service_account_token" "token" {
+  target_service_account = "%s"
+  scopes                = ["https://www.googleapis.com/auth/cloud-platform"]
+}
+`, serviceAccountEmail)
+}
+
+func testAccEphemeralServiceAccountToken_withDelegates(serviceAccountEmail string) string {
+	return fmt.Sprintf(`
+ephemeral "google_service_account_token" "token" {
+  target_service_account = "%s"
+  delegates             = ["%[1]s"]
+  lifetime             = "1200s"
+  scopes               = ["https://www.googleapis.com/auth/cloud-platform"]
+}
+`, serviceAccountEmail)
+}
```

Comment:
Background: [A delegation chain](https://cloud.google.com/iam/docs/create-short-lived-credentials-delegated#sa-credentials-permissions) requires there to be 3 service accounts in use:
- (SA_1) The 'calling' service account that is the identity that wants to get some short-lived credentials
    - This is going to be the identity that Terraform uses, i.e. the service account that your local SA key file is linked to.
- (SA_2) The intermediate or 'delegate' service account that is/are the go-between. This is where a delegation chain is formed and there could be >1 delegate.
    - `delegates` would be a list of 1+ service accounts to make this chain, similar to  
- (SA_3) The 'target' service account which we want to obtain credentials for.

For the delegation chain to work SA_1 needs permissions on SA_2 and SA_2 needs permissions on SA_3 (see the docs I linked).

---

`testAccEphemeralServiceAccountToken_withDelegates` is making the delegate SA the same as the target SA, which is confusing and I think means we are not testing delegation accurately.

Could you please update the test so that `target_service_account` and `delegates` reference separate service accounts? [This might be a useful reference of setting up delegation in an acceptance test.](https://github.com/GoogleCloudPlatform/magic-modules/blob/8ba278ec6b9c43e8d0a65211cc40d4d02168cb91/mmv1/third_party/terraform/provider/provider_impersonate_service_account_delegates_test.go#L132)

Reply:
I think in TestEphemeralServiceAccountToken_withDelegates you need something like this:

```diff
	serviceAccount := envvar.GetTestServiceAccountFromEnv(t)
+	delegatesServiceAccountEmail := acctest.BootstrapServiceAccount(t, "delegate", serviceAccount) // SA_2
+	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "target", delegatesServiceAccountEmail) // SA_3
-	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "delegates", serviceAccount)
```

So you have the 3 service accounts set up, with SA_2 having permissions to make tokens for SA_3